### ISSUE TITLE
Warnings are treated as errors in CI builds but not VS/Command-lines builds

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#6764]
Building inside Visual Studio/command-line should result in the same warnings/errors as via the CIBuild. 

I hit this with this PR: https://github.com/dotnet/roslyn/pull/6756. There are currently 25 warnings in these projects. These show up as warnings both via the command-line and inside VS, but turn into errors in the CIBuild. This shouldn't happen, builds should be exactly the same regardless of how I kicked it off.

The reason we were checking in this code was warnings was because this is a port of existing code, we were planning on cleaning them up in a followup PR as it required thought/work to figure out the correct way to fix them.

**Comment by davkean** [comment]
tag @dotnet/roslyn-infrastructure 

Apparently @tmat had push back against this, so I'd like to understand this before we make this change.

**Comment by tmat** [comment]
@davkean I'd be fine getting errors when binding on command line, but not in VS for some class of warnings.

Warnings like unreachable code or code style warnings shouldn't fail VS build. When I'm working on some code I need to temporarily change the code in non-so-clean way that produces warnings. I still want to be able to build and debug such code. Only after I'm done figuring it out I clean it up and then I should get errors.

**Comment by davkean** [comment]
I don't like this behavior at all, it feels like this should be a per-user setting and not forced on everyone. @jaredpar Opinion?

**Comment by sharwell** [comment]
Treating warnings as errors inside the IDE hurts developer inner loop efficiency. <kbd>F5</kbd> to debug should work as early and often as possible. Do you still want to keep this open for reviewing the command line behavior?
